src/main/java/nextstep/courses/service/SessionService.java

```diff
@@ -26,8 +27,12 @@ public Optional<Sessions> sessions(Long courseId){
         return sessionRepository.findByCourseId(courseId);
     }
 
-    public void enroll(NsUser user, SessionDTO sessionDTO){
-        enrollmentRepository.save(user.getId(), sessionDTO.getId());
-        sessionRepository.save(sessionDTO);
+    public void enroll(NsUser user, Long sessionId){
+        NsUsers nsUsers = enrollmentRepository.findBySessionId(sessionId);
+        Session session = sessionRepository.findById(sessionId);
+        session.replaceEnrollmentNsUsers(e->nsUsers);
+        session.enroll(user);
```

Comment:
```suggestion
        Enrollment enrollment = session.enrollment(nsUsers);
        Student student = enrollment.enroll(user);
```

위와 같이 Session은 수강 신청 로직을 담당하는 Enrollment와 같은 객체를 생성하고, Enrollment가 수강 신청을 로직을 구현하도록 하는 것은 어떨까?